boards/rpi-pico/doc.txt

```diff
@@ -0,0 +1,138 @@
+is running from via XIP, the switch is difficult to read out from software. This is currently not
+supported.
+
+### Pinout
+
+![Pinout Diagram of RPi Pico](https://projects-static.raspberrypi.org/projects/getting-started-with-the-pico/f009ad94826c2f0cd7573a295897e76955301096/en/images/Pico-R3-Pinout.png)
+
+## Flashing the Board
+
+### Flashing the Board Using OpenOCD
+
+Currently (June 2021), only two methods for debugging via OpenOCD are supported:
+
+1. Using a bit-banging low-level adapter, e.g. via the GPIOs of a Raspberry Pi 4B
+2. Using a virtual CMSIS-DAP adapter provided by the second CPU core via
+   https://github.com/majbthrd/pico-debug
+
+Since option 2 requires no additional hardware, this is currently the default. However, you need to
+first "flash" the gimme-cache variant of [pico-debug](https://github.com/majbthrd/pico-debug)
+into RAM using the UF2 bootloader. This will result in the Raspberry Pi Pico showing up as
+CMSIS-DAP debugger. Afterwards run:
+
+```
+make BOARD=rpi-pico flash
+```
+
+***Beware:*** The `rpi-pico` virtual debugger is not persistent and needs to be "flashed" into RAM
+again after each cold boot.
+
+### Flashing the Board Using the Bootloader
+
+Connect the device to your Micro-USB cable while the button (labeled `BOOTSEL` on the silkscreen
+of the PCB) is pressed to enter the bootloader. Afterwards run:
+
+```
+make BOARD=rpi-pico PROGRAMMER=uf2conv flash
+```
+
+### Flashing the Board Using J-Link
+
+Connect the Board to an Segger J-Link debugger, e.g. the EDU mini debugger is relatively affordable,
+but limited to educational purposes. Afterwards run:
+
+```
+make BOARD=rpi-pico PROGRAMMER=jlink flash
+```
+
+## On-Chip Debugging
+
+There are currently (June 2021) few hardware options for debugging the Raspberry Pi Pico:
+
+1. Via J-Link using one of Seggers debuggers
+2. Via OpenOCD using a low-level bit-banging debugger (e.g. a Raspberry Pi 4B with the GPIOs
+   connected to the Raspberry Pi Pico via jump wires)
+3. Via a recently updated [Black Magic Probe](https://github.com/blacksphere/blackmagic)
+
+In addition, a software-only option is possible using
+[pico-debug](https://github.com/majbthrd/pico-debug). The default linker script reserved 16 KiB of
+RAM for this debugger, hence just "flash" the "gimme-cache" flavor into RAM using the UF2
+bootloader. Once this is done, debugging is as simple as running:
+
+```
+make BOARD=rpi-pico debug
+```
+
+***Beware:*** The `rpi-pico` virtual debugger is not persistent and needs to be "flashed" into RAM
+again after each cold boot. The initialization code of RIOT now seems to play well with the
+debugger, so it remains persistent on soft reboots. If you face issues with losing connection to
+the debugger on reboot, try `monitor reset init` in GDB to soft-reboot instead.
+
+## Known Issues / Problems
+
+### Early state Implementation
+
+Currently no support for the following peripherals is implemented:
```

Comment:
we should add the second CPU core to this list

Reply:
> What's the output of `tests/periph_gpio`'s `auto_test`?

```
$ make BOARD=rpi-pico flash test -C tests/periph_gpio
make: Entering directory '/home/maribu/Repos/software/RIOT/tests/periph_gpio'
There are unsatisfied feature requirements: periph_timer
/home/maribu/Repos/software/RIOT/tests/periph_gpio/../../Makefile.include:934: *** You can let the build continue on expected errors by setting CONTINUE_ON_EXPECTED_ERRORS=1 to the command line.  Stop.
make: Leaving directory '/home/maribu/Repos/software/RIOT/tests/periph_gpio'
```

:-/

Reply:
Done